src/sentry/monitors/tasks.py

```diff
@@ -221,10 +222,40 @@ def mark_environment_missing(monitor_environment_id: int, ts: datetime):
         expected_time=expected_time,
         monitor_config=monitor.get_validated_config(),
     )
-    # TODO(epurkhiser): To properly fix GH-55874 we should NOT pass
-    # timezone.now, we need to use the reference timestamp. But I'm not 100%
-    # sure what that should look like yet.
-    mark_failed(checkin, ts=timezone.now())
+
+    # Compute when the check-in *should* have happened given the current
+    # reference timestamp. This is different from the expected_time usage above
+    # as it is computing that most recent expected check-in time using our
+    # reference time. `expected_time` is when the check-in was expected to
+    # happen. This takes advantage of the fact that the current reference time
+    # will always be at least a minute after the last expected check-in.
+    #
+    # Typically `expected_time` and this calculate time should be the same, but
+    # there are cases where it may not be:
+    #
+    #  1. We are guarding against a task not having not run for every minute.
```

Comment:
is this `not having not run` correct here?

Reply:
Good catch, will fix, it should be "Not having run"